integrations/extensions/starter-kits/language-model-conversational-search/elasticsearch-watsonx-actions.json

```diff
@@ -84,13 +84,25 @@
               "variables": [
                 {
                   "value": {
-                    "expression": "{\n    \"text_expansion\": {\n      \"ml.tokens\": {\n        \"model_id\": $es_model,\n        \"model_text\": $query_text\n      }\n    }\n  }"
+                    "expression": "{\n    \"nested\": {\n      \"path\": \"passages\",\n      \"query\": { \n        \"text_expansion\": {\n          \"passages.sparse.tokens\": {\n            \"model_id\": \".elser_model_1\",\n            \"model_text\": ${query_text}\n          }\n        }\n      },\n      \"inner_hits\": {\"_source\": {\"excludes\": [\"passages.sparse\"]}}\n    }\n}"
+                  },
+                  "skill_variable": "query_body_nested"
+                },
+                {
+                  "value": {
+                    "expression": "{\n    \"text_expansion\": {\n      \"ml.tokens\": {\n        \"model_id\": $es_model,\n        \"model_text\": ${query_text}\n      }\n    }\n  }"
+                  },
+                  "skill_variable": "query_body_generic"
+                },
+                {
+                  "value": {
+                    "expression": "${has_inner_hits} ? ${query_body_nested} : ${query_body_generic}"
                   },
                   "skill_variable": "query_body"
                 },
                 {
                   "value": {
-                    "expression": "[\"title\", \"text\"]"
+                    "expression": "${has_inner_hits} ? [\"title\", \"url\", \"text\"] : [\"title\", \"text\"]"
```

Comment:
Just curious as to why this was needed? It doesn't seem like the `url` field from the source is used anywhere downstream to pass to generate answers or another action , so it may be ok to leave this field blank and direct the user to manipulate it as needed based on what is in their index?

Reply:
Thanks for the comment, Sneha. You are right that `url` is not used anywhere and it is only relevant to the web crawler. I am using the source here mostly to limit the fields returned in the search results. Without specifying the source fields, some large and unnecessary fields like `body_content` will be included in each search result. The expectation is that users can manipulate it as they get more familiar with ES and their index structure.

Reply:
I understand that large fields will be returned without adding the source fields. However, my suggestion was to leave it as-is without adding a special condition for `has_inner_hits` as the `url` was not being referenced later on

Reply:
I see. I think the `query_source` was initially defined in this way in your workspace JSON, so I just updated the fields for the case when `has_inner_hits` is true. But it makes sense to leave it as ["title", "text"].